theano/tensor/tests/test_extra_ops.py

```diff
@@ -758,6 +758,137 @@ def test_infer_shape_matrix(self):
+            ops = [self.op_class(*args, **kwargs) for args, kwargs in
+                   self.ops_pars]
+            inp = np.asarray([2, 1, 3, 2], dtype=config.floatX)
+            list_outs_expected = [[np.unique(inp, **kwargs)] if len(args) == 0
+                                  else np.unique(inp, *args, **kwargs)
+                                  for args, kwargs in self.ops_pars]
+            for op, outs_expected in zip(ops, list_outs_expected):
+                f = theano.function(inputs=[x],
+                                    outputs=op(x, return_list=True))
+                outs = f(inp)
+                # Compare the result computed to the expected value.
+                for out, out_exp in zip(outs, outs_expected):
+                    utt.assert_allclose(out, out_exp)
+        else:
+            raise utt.SkipTest('Requires numpy >= 1.13')
+
+    def test_basic_matrix(self):
+        if self.expect_success:
+            # Basic test for a matrix.
+            # Done by using the op and checking that it returns the right
+            # answer.
+
+            x = theano.tensor.matrix()
+            ops = [self.op_class(*args, **kwargs) for args, kwargs in
+                   self.ops_pars]
+            inp = np.asarray([[2, 1], [3, 2], [2, 1]], dtype=config.floatX)
+            list_outs_expected = [[np.unique(inp, **kwargs)] if len(args) == 0
+                                  else np.unique(inp, *args, **kwargs)
+                                  for args, kwargs in self.ops_pars]
+            for op, outs_expected in zip(ops, list_outs_expected):
+                f = theano.function(inputs=[x],
+                                    outputs=op(x, return_list=True))
+                outs = f(inp)
+                # Compare the result computed to the expected value.
+                for out, out_exp in zip(outs, outs_expected):
+                    utt.assert_allclose(out, out_exp)
+        else:
+            raise utt.SkipTest('Requires numpy >= 1.13')
+
+    def test_infer_shape_vector(self):
+        if self.expect_success:
+            # Testing the infer_shape with a vector.
+
+            x = theano.tensor.vector()
+
+            ops = [self.op_class(*args, **kwargs) for args, kwargs in
+                   self.ops_pars]
+            for op in ops:
+                if not op.return_inverse:
+                    continue
+                if op.return_index:
+                    f = op(x)[2]
+                else:
+                    f = op(x)[1]
+                self._compile_and_check([x],
+                                        [f],
+                                        [np.asarray(np.array([2, 1, 3, 2]),
+                                                    dtype=config.floatX)],
+                                        self.op_class)
+        else:
+            raise utt.SkipTest('Requires numpy >= 1.13')
+
+    def test_infer_shape_matrix(self):
+        if self.expect_success:
```

Comment:
Personal preference. Feel free to ignore.

I perfer this construction. It limit the indentation level of the code. So it make it more clear. I also find that more clear to have the small else case first. Here is what I prefer:

if not self.expect_success:
   raise ...
...

The long clause is one less indentation and we see more clearly the small clause at the beginning.

Reply:
I like your suggestion. I'll implement it that way.